gromacs/config.py

```diff
@@ -309,23 +308,15 @@
 def _generate_template_dict(dirname):
     """Generate a list of included files *and* extract them to a temp space.
 
-    Templates have to be extracted from the egg because they are used
-    by external code. All template filenames are stored in
-    :data:`config.templates`.
+    Templates have to be extracted because they are used by external code.
+    All template filenames are stored in config.templates.
     """
-    return dict(
-        (resource_basename(fn), resource_filename(__name__, dirname + "/" + fn))
-        for fn in resource_listdir(__name__, dirname)
-        if not fn.endswith("~")
-    )
-
-
-def resource_basename(resource):
-    """Last component of a resource (which always uses '/' as sep)."""
-    if resource.endswith("/"):
-        resource = resource[:-1]
-    parts = resource.split("/")
-    return parts[-1]
+    resource_path = Path(__package__) / dirname
```

Comment:
This does not necessarily work with installed packages, I think, or at least if these are zipped or eggs or whatever.

https://importlib-resources.readthedocs.io/en/latest/using.html#file-system-or-zip-file and in particular the `atexit` solution in https://importlib-resources.readthedocs.io/en/latest/migration.html#pkg-resources-resource-filename

Probably needs something like
```python
import importlib.resources as importlib_resources
from contextlib import ExitStack
import atexit

file_manager = ExitStack()
atexit.register(file_manager.close)
resource_path = importlib_resources.files(gromacs.config) / dirname
path = file_manager.enter_context(importlib_resources.as_file(resource_path))
```